.github/workflows/ci.yml

```diff
@@ -40,6 +41,14 @@ jobs:
         with:
           version: nightly
 
+      - name: Cache fork requests
+        uses: actions/cache@v3
+        with:
+          path: ~/.foundry/cache
+          key: ${{ runner.os }}-foundry-network-fork-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-foundry-network-fork-
```

Comment:
This who step using `actions/cache@v3` shouldn't be needed, `foundry-rs/foundry-toolchain@v1` now caches automatically

Reply:
Hm, I removed this in d7b4570 and caching stopped working

For example, you can see that in [this build](https://github.com/gitcoinco/2022-Governor-upgrade/actions/runs/3933991543/attempts/2) that tests ran to completion, taking 9 minutes. I figured this was expected since we were using a new caching approach -- we'd need to re-cache everything from scratch. So after the build completed I immediately re-ran the workflow to see if things would be faster now. They weren't. I killed [the next build](https://github.com/gitcoinco/2022-Governor-upgrade/actions/runs/3933991543/attempts/3) after 3 minutes.

For comparison, prior to removing this explicit cache step, fork tests with cached RPC calls were finishing up in [~40s](https://github.com/gitcoinco/2022-Governor-upgrade/actions/runs/3933933982/attempts/2)

Reply:
Hmm yea that's odd. I just ran the same workflow twice:
1. https://github.com/gitcoinco/2022-Governor-upgrade/actions/runs/3934573618/jobs/6729444859
2. https://github.com/gitcoinco/2022-Governor-upgrade/actions/runs/3934573618/jobs/6729501579

Both took about 4 minutes. I wonder if [this line](https://github.com/foundry-rs/foundry-toolchain/blob/df3156cc4940f179d199fb27f9e0fbd97a5b82cb/src/cache.js#L18-L20) missing logic to create the cache dir if it doesn't exist, so it's never getting saved the first time? Or perhaps we're doing something wrong. Gonna pull in @paulrberg to make sure we're not missing anything

Reply:
Another test:
1. Pushed adffed2 which creates the cache directory manually in CI
2. CI ran here: https://github.com/gitcoinco/2022-Governor-upgrade/actions/runs/3940835577
3. Manually ran another after here: https://github.com/gitcoinco/2022-Governor-upgrade/actions/runs/3940835577

Both took about the same amount of time 🤔

Reply:
Thanks for the tag, @mds1. Taking a look now.

There might be a bug in the cache implementation - sorry for that. I'm basing this guess on a number of other reports on Twitter.

However, I don't thin't think that this is about the if statement you linked to, i.e. this:

```js
if (fs.existsSync(CACHE_PATHS[0]))
```

I added that in https://github.com/foundry-rs/foundry-toolchain/pull/21 to prevent another bug that occurred when the user didn't have any fork tests.

Reply:
I reverted back to the old approach. I'll leave this thread open for now in case Paul gets back to us, and will plan to merge without resolving.

Reply:
No worries, David.

After a few hours of debugging (and hundreds of console logs), I have managed to fix this. Waiting for @mds1 or someone else from the team to review my PR and apply the fix:

https://github.com/foundry-rs/foundry-toolchain/pull/24

I am quite confident that the caching should work as expected now!